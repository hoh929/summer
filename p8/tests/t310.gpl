int window_width = 200;
int window_height = 200;
int first_stop = 1;
int animation_steps = 0;

forward animation bounce(circle cur_circle);
forward animation stop(circle cur_circle2);

circle ball(x = 100, y = 100, animation_block = bounce);
int ball_x_increment = 5;
int ball_y_increment = 2;
int ball_size = 10;

initialization
{
  print("This program should quit on its own.  Your program has a bug if this does not quit on its own.");
}


animation bounce(circle cur_circle)
{
	// if ball has reached either the left or right, reverse its direction
	if (cur_circle.x < 0 || cur_circle.x > window_width - ball_size)
		ball_x_increment = -ball_x_increment;

	// if ball has reached either the top or bottom, reverse its direction
	if (cur_circle.y < 0 || cur_circle.y > window_height - ball_size)
		ball_y_increment = -ball_y_increment;

	// on each step of the animation, move the ball
	cur_circle.x += ball_x_increment;
	cur_circle.y += ball_y_increment;

  animation_steps += 1;
  if (animation_steps == 100)
  {
    print("this line should only be printed ONCE *********************");
    // this entire test is just to make sure the following works.
    cur_circle.animation_block = stop;
  }
}

animation stop(circle cur_circle2)
{
  cur_circle2.x = 100;
  cur_circle2.y = 100;
  cur_circle2.red = 1;
  cur_circle2.green = 0;
  cur_circle2.blue = 0;
  print("animation stop(circle cur_circle2) called for first and last time");
  exit(42);
}

termination
{
  if (ball.x == 100 && ball.y == 100 && ball.red == 1 && ball.green == 0 && ball.blue == 0)
    print("ball values are correct");
  else
    print("ball values are NOT correct");
    
}
