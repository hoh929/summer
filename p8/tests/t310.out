gpl.cpp::main()
  input file(tests/t310.gpl)
  random seed(42)
  read_keypresses_from_standard_input(false)
  dump_pixels(false)
  symbol_table(true)
  graphics(true)

gpl.cpp::main() Calling yyparse()


gpl.cpp::main() after call to yyparse().


Printing the symbol table from main()
int animation_steps = 0
circle ball
{
    animation_block = bounce,
    blue = double(1),
    drawing_order = int(0),
    green = double(0),
    h = int(20),
    proximity = int(4),
    radius = int(10),
    red = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(20),
    x = int(100),
    y = int(100)
}
int ball_size = 10
int ball_x_increment = 5
int ball_y_increment = 2
animation_block bounce
circle cur_circle
{
    animation_block = NULL,
    blue = double(1),
    drawing_order = int(0),
    green = double(0),
    h = int(20),
    proximity = int(4),
    radius = int(10),
    red = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(20),
    x = int(0),
    y = int(0)
}
circle cur_circle2
{
    animation_block = NULL,
    blue = double(1),
    drawing_order = int(0),
    green = double(0),
    h = int(20),
    proximity = int(4),
    radius = int(10),
    red = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(20),
    x = int(0),
    y = int(0)
}
int first_stop = 1
animation_block stop
int window_height = 200
int window_width = 200
gpl.cpp::main() Calling window->initialize().

print[16]: This program should quit on its own.  Your program has a bug if this does not quit on its own.

gpl.cpp::main() Passing control to window->main_loop().
print[37]: this line should only be printed ONCE *********************
print[50]: animation stop(circle cur_circle2) called for first and last time
gpl[51]: exit(42)
print[57]: ball values are correct
